animation_nodes/nodes/mesh/triangulate_mesh.py

```diff
@@ -0,0 +1,35 @@
+import bpy
+from bpy.props import *
+from ... events import propertyChanged
+from ... base_types import AnimationNode
+from ... data_structures.meshes.validate import checkMeshData
+
+class TriangulateMeshNode(bpy.types.Node, AnimationNode):
+    bl_idname = "an_TriangulateMeshNode"
+    bl_label = "Triangulate Mesh"
+    errorHandlingType = "EXCEPTION"
+
+    methodType: BoolProperty(name = "Use Advanced Method", default = False,
+                             description = "Use advanced method for Mesh Triangulation",
+                             update = propertyChanged)
+
+    def create(self):
+        self.newInput("Mesh", "Mesh", "mesh")
+
+        self.newOutput("Mesh", "Mesh", "meshOut")
+
+    def draw(self, layout):
+        layout.prop(self, "methodType", text = "Use Advanced Method")
+
+    def execute(self, mesh):
+        meshOut = mesh.copy()
```

Comment:
Use `dataIsModified` for the input mesh.

Reply:
That gives this error,
![Screenshot from 2020-04-22 17-28-49](https://user-images.githubusercontent.com/30294746/79979232-d2973900-84be-11ea-8800-911ca02336d5.png)

Reply:
Can you show me the relevant code using `dataIsModified`?

Reply:
This is edited _Traingulate Mesh_ node code,
```
import bpy
from bpy.props import *
from ... events import propertyChanged
from ... base_types import AnimationNode
from ... data_structures.meshes.validate import checkMeshData

class TriangulateMeshNode(bpy.types.Node, AnimationNode):
    bl_idname = "an_TriangulateMeshNode"
    bl_label = "Triangulate Mesh"
    errorHandlingType = "EXCEPTION"

    methodType: BoolProperty(name = "Use Advanced Method", default = False,
                             description = "Use advanced method for Mesh Triangulation",
                             update = propertyChanged)

    def create(self):
        self.newInput("Mesh", "Mesh", "mesh", dataIsModified = True)

        self.newOutput("Mesh", "Mesh", "meshOut")

    def draw(self, layout):
        layout.prop(self, "methodType", text = "Use Advanced Method")

    def execute(self, mesh):
        if self.methodType:
            mesh.triangulateMesh(method = "EAR")
        else:
            mesh.triangulateMesh(method = "FAN")

        try:
            checkMeshData(mesh.vertices, mesh.edges, mesh.polygons)
            return mesh
        except Exception as e:
            self.raiseErrorMessage(str(e))
```

Reply:
This is the function in the `mesh_data.pyx` file
https://github.com/JacquesLucke/animation_nodes/blob/e2195df285ff9182e3b1cadbf7624ed9bfa8da9a/animation_nodes/data_structures/meshes/mesh_data.pyx#L181-L188

Reply:
I am not sure. I will have to investigate this myself. So just ignore it for now.

Reply:
Okay.